programs/ssl/ssl_server2.c

```diff
@@ -3742,6 +3745,12 @@ int main( int argc, char *argv[] )
     }
 #endif /* MBEDTLS_SSL_DTLS_CONNECTION_ID */
 
+#if defined(MBEDTLS_MEMORY_DEBUG)
+    mbedtls_memory_buffer_alloc_cur_get(&current_heap_memory, &heap_blocks);
+    mbedtls_memory_buffer_alloc_max_get(&peak_heap_memory, &heap_blocks);
+    mbedtls_printf("Heap memory usage after handshake: %zd bytes. In peak was %zd\n", current_heap_memory, peak_heap_memory);
```

Comment:
We may not be able to use `%zd` on all platforms we want this program to run on. Consider printing in a more portable way if CI indicates this is necessary.

Reply:
changed to %u